src/storage/fileio/mod.rs

```diff
@@ -0,0 +1,167 @@
+use std::{
+    fs::{File, OpenOptions},
+    io,
+};
+
+#[cfg(target_family = "unix")]
+use std::os::unix::prelude::AsRawFd;
+
+#[cfg(target_family = "unix")]
+use std::os::unix::prelude::FileExt;
+
+pub struct FileIO {}
```

Comment:
You can end this line off with a semicolon just to show that it contains no data (i.e. `pub struct FileIO;`).